src/wire/ip.rs

```diff
@@ -126,6 +127,55 @@ impl fmt::Display for Address {
     }
 }
 
+/// A CIDR containing a prefix of an IP subnet.
+pub enum Cidr {
+    Ipv4(Ipv4Cidr),
+}
+
+impl Cidr {
+    /// Construct a new CIDR
+    pub fn new(addr: Address, mask_len: u8) -> Cidr {
```

Comment:
This function doesn't seem very useful as the valid ranges of prefix lengths depend on the address.

Reply:
I've rephrased it so the constructors check prefix_len and can fail now.